armi/reactor/converters/axialExpansion/tests/buildAxialExpAssembly.py

```diff
@@ -0,0 +1,142 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from armi import materials
+from armi.utils import units
+from armi.reactor.assemblies import HexAssembly, grids
+from armi.reactor.blocks import HexBlock
+from armi.reactor.components.basicShapes import Circle, Hexagon
+from armi.reactor.components import DerivedShape
+
+
+def buildTestAssembly(materialName: str, hot: bool = False):
+    """Create test assembly.
+
+    Parameters
+    ----------
+    materialName: string
+        determines which material to use
+    hot: boolean
+        determines if assembly should be at hot temperatures
+    """
+    if not hot:
+        hotTemp = 25.0
+        height = 10.0
+    else:
+        hotTemp = 250.0
+        height = 10.0 + 0.02 * (250.0 - 25.0)
+
+    assembly = HexAssembly("testAssemblyType")
+    assembly.spatialGrid = grids.axialUnitGrid(numCells=1)
+    assembly.spatialGrid.armiObject = assembly
+    assembly.add(buildTestBlock("shield", materialName, hotTemp, height))
+    assembly.add(buildTestBlock("fuel", materialName, hotTemp, height))
+    assembly.add(buildTestBlock("fuel", materialName, hotTemp, height))
+    assembly.add(buildTestBlock("plenum", materialName, hotTemp, height))
+    assembly.add(buildDummySodium(hotTemp, height))
+    assembly.calculateZCoords()
+    assembly.reestablishBlockOrder()
+    return assembly
+
+
+def buildTestBlock(blockType: str, materialName: str, hotTemp: float, height: float):
+    """Return a simple pin type block filled with coolant and surrounded by duct.
+
+    Parameters
+    ----------
+    blockType : string
+        determines which type of block you're building
+    materialName : string
+        determines which material to use
+    """
+    b = HexBlock(blockType, height=height)
+
+    fuelDims = {"Tinput": 25.0, "Thot": hotTemp, "od": 0.76, "id": 0.00, "mult": 127.0}
+    cladDims = {"Tinput": 25.0, "Thot": hotTemp, "od": 0.80, "id": 0.77, "mult": 127.0}
+    ductDims = {"Tinput": 25.0, "Thot": hotTemp, "op": 16, "ip": 15.3, "mult": 1.0}
+    intercoolantDims = {
+        "Tinput": 25.0,
+        "Thot": hotTemp,
+        "op": 17.0,
+        "ip": ductDims["op"],
+        "mult": 1.0,
+    }
+    coolDims = {"Tinput": 25.0, "Thot": hotTemp}
+    mainType = Circle(blockType, materialName, **fuelDims)
+    clad = Circle("clad", materialName, **cladDims)
+    duct = Hexagon("duct", materialName, **ductDims)
+
+    coolant = DerivedShape("coolant", "Sodium", **coolDims)
+    intercoolant = Hexagon("intercoolant", "Sodium", **intercoolantDims)
+
+    b.add(mainType)
+    b.add(clad)
+    b.add(duct)
+    b.add(coolant)
+    b.add(intercoolant)
+    b.setType(blockType)
+
+    b.getVolumeFractions()
+
+    return b
+
+
+def buildDummySodium(hotTemp: float, height: float):
```

Comment:
For some reason, this name seems too short to me. Maybe add "Block"?

> buildDummySodiumBlock

Thoughts?

Reply:
I'm not sure if it's documented somewhere, but maybe explaining the significance of a `DUMMY` name would be valuable as a test with the flag system. Essentially you might want a test that demonstrates this method is returning the expected thing if not already checked by itself in a test just in case this flag gets broken later by accident.